backend/de.metas.cucumber/src/test/resources/de/metas/cucumber/features/shipment/shipmentDeliveryDateRule.feature

```diff
@@ -0,0 +1,240 @@
+    And metasfresh contains M_ProductPrices
+      | Identifier | M_PriceList_Version_ID.Identifier | M_Product_ID.Identifier | PriceStd | C_UOM_ID.X12DE355 | C_TaxCategory_ID.InternalName |
+      | pp_1       | plv_1                             | p_1                     | 10.0     | PCE               | Normal                        |
+    And metasfresh contains C_BPartners:
+      | Identifier    | Name                   | OPT.IsVendor | OPT.IsCustomer | M_PricingSystem_ID.Identifier |
+      | endcustomer_1 | Endcustomer18032022_12 | N            | Y              | ps_1                          |
+    And metasfresh contains C_BPartner_Locations:
+      | Identifier | GLN           | C_BPartner_ID.Identifier | OPT.IsShipToDefault | OPT.IsBillToDefault |
+      | l_1        | 0123456789011 | endcustomer_1            | Y                   | Y                   |
+
+
+  @Id:S0XXX.1_100
+  @from:cucumber
+  Scenario: Shipment DeliveryDateRule on generate shipments - PreparationDate as movementDate
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered | OPT.C_PaymentTerm_ID | OPT.PreparationDate     | OPT.POReference        |
+      | o_1        | true    | endcustomer_1            | 2021-04-16  | 1000012              | 2021-04-18T21:00:00.00Z | DeliveryDateRuleTest_1 |
+    And metasfresh contains C_OrderLines:
+      | Identifier | C_Order_ID.Identifier | M_Product_ID.Identifier | QtyEntered |
+      | ol_1       | o_1                   | p_1                     | 10         |
+    When the order identified by o_1 is completed
+    And after not more than 60s, M_ShipmentSchedules are found:
+      | Identifier | C_OrderLine_ID.Identifier | IsToRecompute |
+      | s_s_1      | ol_1                      | N             |
+    And 'generate shipments' process is invoked
+      | M_ShipmentSchedule_ID.Identifier | QuantityType | IsCompleteShipments |
+      | s_s_1                            | D            | Y                   |
+    And after not more than 60s, M_InOut is found:
+      | M_ShipmentSchedule_ID.Identifier | M_InOut_ID.Identifier |
+      | s_s_1                            | s_1                   |
+    And validate the created shipments
+      | M_InOut_ID.Identifier | C_BPartner_ID.Identifier | C_BPartner_Location_ID.Identifier | dateordered | processed | docStatus | OPT.MovementDate | OPT.POReference        |
+      | s_1                   | endcustomer_1            | l_1                               | 2021-04-16  | true      | CO        | 2021-04-18       | DeliveryDateRuleTest_1 |
+
+  @Id:S0XXX.1_200
+  @from:cucumber
+  Scenario: Shipment DeliveryDateRule on generate shipments - Today as movementDate if PreparationDate is in the past
+    And metasfresh contains C_Orders:
+      | Identifier | IsSOTrx | C_BPartner_ID.Identifier | DateOrdered | OPT.C_PaymentTerm_ID | OPT.PreparationDate     | OPT.POReference        |
```

Comment:
* use C_BPartner_ID instead of C_BPartner_ID.Identifier 
* AFAIR, u don't have to prefix them with `OPT.`
* OPT.C_PaymentTerm_ID - is it needed?
* OPT.PreparationDate     - is it needed?
* OPT.POReference   - is it needed?

Reply:
as a general rule: try creating the cucumber tests as skinny as possible
* avoid using `.Identifier` 
* avoid using `OPT.`
* avoid providing irelevant data like payment term, reference no etc